client/verta/verta/endpoint/_endpoint.py

```diff
@@ -509,8 +509,15 @@ def _set_kafka_settings(self, stage_id, kafka_settings):
                 " not {}".format(type(kafka_settings))
             )
         if kafka_settings._cluster_config_id is None:
-            config = kafka.list_kafka_configurations(self._conn)[0]
-            kafka_settings._cluster_config_id = config.get("id", None) if config else None
+            configs = kafka.list_kafka_configurations(self._conn)
+            if not configs:
+                raise RuntimeError("no valid Kafka configuration found.")
```

Comment:
Throwing in a little actionable suggestion in the error message:

```suggestion
                raise RuntimeError("no Kafka configuration found; please ensure that Kafka is configured in Verta")
```

Reply:
Also, I'd recommend against having a trailing period in our error messages. I don't think the client does that anywhere else.